src/js/releases.js

```diff
@@ -40,77 +35,62 @@ function buildLatestHTML(releasesJson) {
-
-      // if the filename contains both the platform name and the matching INSTALLER extension, add the relevant info to the asset object
-      ASSETOBJECT.thisInstallerExtension = getInstallerExt(ASSETOBJECT.thisPlatform);
-      ASSETOBJECT.thisBinaryExtension = getBinaryExt(ASSETOBJECT.thisPlatform);
-      if (uppercaseFilename.indexOf(ASSETOBJECT.thisInstallerExtension.toUpperCase()) >= 0) {
-        if (ASSETARRAY.length > 0) {
-          ASSETARRAY.forEach(function (asset) {
-            if (asset.thisPlatform === ASSETOBJECT.thisPlatform) {
-              ASSETARRAY.pop();
-            }
-          });
-        }
-        ASSETOBJECT.thisPlatformExists = true;
-        ASSETOBJECT.thisInstallerExists = true;
-        ASSETOBJECT.thisInstallerLink = eachAsset.binary_link;
-        ASSETOBJECT.thisInstallerSize = Math.floor(eachAsset.binary_size / 1024 / 1024);
-        ASSETOBJECT.thisBinaryExists = true;
-        ASSETOBJECT.thisBinaryLink = eachAsset.binary_link.replace(ASSETOBJECT.thisInstallerExtension, ASSETOBJECT.thisBinaryExtension);
-        ASSETOBJECT.thisBinarySize = Math.floor(eachAsset.binary_size / 1024 / 1024);
-        ASSETOBJECT.thisChecksumLink = eachAsset.binary_link.replace(ASSETOBJECT.thisInstallerExtension, '.sha256.txt');
-      }
-      // if the filename contains both the platform name and the matching BINARY extension, add the relevant info to the asset object
-      if (uppercaseFilename.indexOf(ASSETOBJECT.thisBinaryExtension.toUpperCase()) >= 0) {
-        var installerExist = false;
-        if (ASSETARRAY.length > 0) {
-          ASSETARRAY.forEach(function (asset) {
-            if (asset.thisPlatform === ASSETOBJECT.thisPlatform) {
-              installerExist = true;
-            }
-          });
-        }
-        if (!installerExist) {
-          ASSETOBJECT.thisPlatformExists = true;
-          ASSETOBJECT.thisBinaryExists = true;
-          ASSETOBJECT.thisBinaryLink = eachAsset.binary_link;
-          ASSETOBJECT.thisBinarySize = Math.floor(eachAsset.binary_size / 1024 / 1024);
-          ASSETOBJECT.thisChecksumLink = eachAsset.checksum_link
-        }
-      }
+  // Array of releases that have binaries we want to display
+  var releases = [];
 
-      if (ASSETOBJECT.thisPlatformExists === true) {
-        ASSETARRAY.push(ASSETOBJECT);
-      }
+  releasesJson.forEach((releaseAsset) => {
+    const platform = findPlatform(releaseAsset);
+
+    // Skip this asset if its platform could not be matched (see the website's 'config.json')
+    if (!platform) {
+      return;
+    }
+
+    // Skip this asset if it's not a binary type we're interested in displaying
+    const binary_type = releaseAsset.binary_type.toUpperCase();
+    if (['INSTALLER', 'JDK', 'JRE'].indexOf(binary_type) === -1) {
+      return;
+    }
+
+    // Get the existing release asset (passed to the template) or define a new one
+    var release = releases.find((p) => p.platform_name === platform);
```

Comment:
nitpick - what's a `p`? :-)

Reply:
Oops-- I had "platform" on the mind, apparently.  I've updated this in ca6da266238294642fbb69949f7d568a4c38ff2c.